app/controllers/stages_controller.rb

```diff
@@ -102,4 +99,12 @@ def stage_params
   def find_stage
     @stage = current_project.stages.find_by_param!(params[:id])
   end
+
+  def template_param
+    @template_param ||= ActiveRecord::Type::Boolean.new.type_cast_from_database(stage_params.delete("template"))
+  end
+
+  def update_template
+    template_param ? @stage.project.template!(@stage) : @stage.project.template!
```

Comment:
sorry. mega pr. 

I'll add kwargs. 

There's two modes because, if you simply want to unset the current template, you need a way to do that without specifying a new stage as template.